akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala

```diff
@@ -3,102 +3,57 @@
  */
 package akka.persistence.typed.scaladsl
 
-import akka.actor.typed.Behavior.UntypedBehavior
-import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.Behavior
+import akka.actor.typed.Behavior.DeferredBehavior
+import akka.actor.typed.internal.TimerSchedulerImpl
+import akka.{ actor ⇒ a }
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, TimerScheduler }
 import akka.annotation.{ DoNotInherit, InternalApi }
 import akka.persistence.typed.internal._
+import akka.persistence.typed.scaladsl.PersistentBehaviors.CommandHandler
+import akka.util.ConstantFun
 
-import scala.collection.{ immutable ⇒ im }
+import scala.language.implicitConversions
 
 object PersistentBehaviors {
 
+  // we use this type internally, however it's easier for users to understand the function, so we use it in external api
+  type CommandHandler[Command, Event, State] = (ActorContext[Command], State, Command) ⇒ Effect[Event, State]
+
   /**
    * Create a `Behavior` for a persistent actor.
    */
   def immutable[Command, Event, State](
     persistenceId:  String,
     initialState:   State,
-    commandHandler: CommandHandler[Command, Event, State],
-    eventHandler:   (State, Event) ⇒ State): PersistentBehavior[Command, Event, State] = {
-    // FIXME remove `persistenceIdFromActorName: String ⇒ String` from PersistentBehavior
-    new PersistentBehavior(_ ⇒ persistenceId, initialState, commandHandler, eventHandler,
-      recoveryCompleted = (_, _) ⇒ (),
-      tagger = _ ⇒ Set.empty,
-      snapshotOn = (_, _, _) ⇒ false)
-  }
+    commandHandler: (ActorContext[Command], State, Command) ⇒ Effect[Event, State],
```

Comment:
Can't/shouldn't we use the `CommandHandler` type here?

Reply:
Ok, though actually I was hoping to avoid the alias entirely... I think it's harder to explain that it's not a real type for people exploring the code